src/solvers/refinement/string_builtin_function.h

```diff
@@ -128,9 +128,12 @@ class string_concat_char_builtin_functiont
     return "concat_char";
   }
 
-  exprt add_constraints(string_constraint_generatort &generator) const override
+  string_constraintst
+  constraints(string_constraint_generatort &generator) const override
   {
-    return generator.add_axioms_for_concat_char(result, input, character);
+    auto pair = generator.add_axioms_for_concat_char(result, input, character);
+    pair.second.existential.push_back(equal_exprt(pair.first, return_code));
+    return pair.second;
```

Comment:
I don't understand what we're doing here, and that commit introduces a lot of code duplication. What exactly are we doing?

Reply:
Yes this is a bit boilerplate, I will try to refactor that

Reply:
I've added commits at the end (the "Move add_axioms_*" ones) which show how the `constraints()` codes should look like. I haven't done it for concat and insert yet because it requires getting rid of the other usages of these add_axioms functions first.